tools/run-mypy

```diff
@@ -80,7 +80,16 @@ if args.quick:
 elif args.warn_unreachable:
     extra_args.append("--warn-unreachable")
 
-mypy_args = extra_args + python_files + pyi_files
+# We name our cache directory based on the HEAD of our working branch.
+cache_commit_id = subprocess.Popen(['git', 'rev-parse', 'HEAD'],
+                                   stdout=subprocess.PIPE).stdout.read().decode()
+cache_args = ['--cache-dir=var/mypy-cache-%s' % (cache_commit_id,)]
```

Comment:
Do we really want to the cache to be invalidated on every commit?

Reply:
No, good catch.  That is a suggestion in the [refinements)(https://mypy.readthedocs.io/en/latest/additional_features.html#refinements) sections in the docs. 

I'm just playing around with the artifacts system right now.